cloud_formation/lambda/ingest_populate/ingest_queue_upload.py

```diff
@@ -73,21 +79,30 @@ def handler(args, context):
 
         retry = 3
         while retry > 0:
-            resp = queue.send_messages(Entries=batch)
-            sent += len(resp['Successful'])
-
-            if 'Failed' in resp and len(resp['Failed']) > 0:
-                print("Batch failed to enqueue messages")
-                print("Retries left: {}".format(retry))
-                print("Boto3 send_messages response: {}".format(resp))
-                time.sleep(SQS_RETRY_TIMEOUT)
-
-                ids = [f['Id'] for f in resp['Failed']]
-                batch = [b for b in batch if b['Id'] in ids]
+            try:
+                try_again = False
+                resp = queue.send_messages(Entries=batch)
+                if 'Successful' in resp:
+                    sent += len(resp['Successful'])
+
+                if 'Failed' in resp and len(resp['Failed']) > 0:
+                    try_again = True
+                    print("Batch failed to enqueue messages")
+                    print("Retries left: {}".format(retry))
+                    print("Boto3 send_messages response: {}".format(resp))
+                    # shorten the batch to only those ids that failed.
+                    ids = [f['Id'] for f in resp['Failed']]
+                    batch = [b for b in batch if b['Id'] in ids]
+            except ClientError as err:
+                try_again = True
+                print('ClientError while sending messages: {}  batch: {}'.format(err, batch))
+
+            if try_again:
                 retry -= 1
                 if retry == 0:
                     print("Exhausted retry count, stopping")
-                    raise FailedToSendMessages(batch) # SFN will relaunch the activity
+                    raise FailedToSendMessages(batch)  # SFN will relaunch the activity
+                time.sleep(SQS_RETRY_TIMEOUT)
```

Comment:
15s may be too long to wait given lambda time limits. Might also want to check the context object to see how much time remaining before lambda timeout.

Reply:
How long do you think, 3 seconds?